fapolicy_analyzer/ui/fapd_manager.py

```diff
@@ -227,10 +229,15 @@ def _status(self):
                 return ServiceStatus.FALSE
 
     def capture_online_state(self):
-        logging.debug("capture_online_state()")
-        updated_online_state = ServiceStatus(self._fapd_ref.is_active())
-        self._fapd_prior_online_state = updated_online_state
-        self._fapd_cur_online_state = updated_online_state
+        logging.debug("capture_online_state(): Capturing current online status")
+        if self._fapd_lock.acquire():
```

Comment:
Same here, `with` would be ideal on these locked blocks.

Reply:
Yeah, `with` context management is Ok here. Will do.